tests/ecosystem/upgrade/conftest.py

```diff
@@ -516,6 +543,66 @@ def pre_upgrade_pods_running_io(
     return pre_upgrade_filesystem_pods + pre_upgrade_block_pods
 
 
+@pytest.fixture(scope='session')
+def fio_configmap_dict_mcg(fio_configmap_dict_session, fio_job_dict_mcg):
+    """
+    Fio configmap dictionary with configuration set for MCG workload.
+    """
+    configmap = copy.deepcopy(fio_configmap_dict_session)
+    config_name = f"{fio_job_dict_mcg['metadata']['name']}-config"
+    configmap['metadata']['name'] = config_name
+    return configmap
+
+
+@pytest.fixture(scope='session')
+def mcg_workload_job(
+    fio_job_dict_mcg,
+    fio_configmap_dict_mcg,
+    fio_conf_mcg,
+    fio_project,
+    tmp_path,
+    request
+):
+    """
+    Creates kubernetes job that should utilize MCG during upgrade.
+
+    Returns:
+        object: Job object
+
+    """
+    fio_configmap_dict_mcg["data"]["workload.fio"] = fio_conf_mcg
+    fio_objs = [fio_configmap_dict_mcg, fio_job_dict_mcg]
+
+    job_name = fio_job_dict_mcg['metadata']['name']
+
+    log.info(f"Creating job {job_name}")
+    job_file = ObjectConfFile(
+        "fio_continuous",
+        fio_objs,
+        fio_project,
+        tmp_path
+    )
+
+    # deploy the Job to the cluster and start it
+    job_file.create()
+    log.info(f"Job {job_name} created")
+
+    # get job object
+    ocp_job_obj = ocp.OCP(kind=constants.JOB, namespace=fio_project.namespace)
```

Comment:
are u using this object anywhere outside the teardown function? if not, may be better to create it within function scope

Reply:
The fixture returns job object which is used in 2 tests: test_start_upgrade_mcg_io and test_upgrade_mcg_io and is kept active during upgrade.

Reply:
my question is, why not create `ocp_job_obj` within fixture scope

Reply:
This is just a wrapper to access `get` function implemented in our framework. It does not create anything so I don't think that it needs to be accessible from outside as it's own fixture. Also, it is not possible to use function scope fixture in session scope fixture.